examples/libf3d/web/src/main.js

```diff
@@ -0,0 +1,311 @@
+import f3d from "f3d";
+
+const settings = {
+  canvas: document.getElementById("canvas"),
+  setupOptions: (options) => {
+    // background must be set to black for proper blending with transparent canvas
+    options.set_color("render.background.color", 0, 0, 0);
+
+    // setup coloring
+    options.toggle("model.scivis.enable");
+    options.set_string("model.scivis.array_name", "Colors");
+    options.set_integer("model.scivis.component", -2);
+    options.toggle("model.scivis.cells");
+
+    // make it look nice
+    options.toggle("render.effect.antialiasing.enable");
+    options.toggle("render.effect.tone_mapping");
+    options.toggle("render.effect.ambient_occlusion");
+    options.toggle("render.hdri.ambient");
+
+    // display widgets
+    options.toggle("ui.axis");
+    options.toggle("render.grid.enable");
+
+    // default to +Z
+    options.set_as_string("scene.up_direction", "+Z");
+  },
+};
+
+f3d(settings)
+  .then(async (Module) => {
+    // write in the filesystem
+    const defaultFile = await fetch("f3d.vtp").then((b) => b.arrayBuffer());
+    Module.FS.writeFile("f3d.vtp", new Uint8Array(defaultFile));
+
+    // automatically load all supported file format readers
+    Module.Engine.autoloadPlugins();
+
+    Module.engineInstance = Module.Engine.create();
+
+    const openFile = (name) => {
+      document.getElementById("file-name").innerHTML = name;
+      const filePath = "/" + name;
+      const scene = Module.engineInstance.getScene();
+      if (scene.supports(filePath)) {
+        scene.clear();
+        scene.add(filePath);
+      } else {
+        console.error("File " + filePath + " cannot be opened");
+      }
+      Module.engineInstance.getWindow().resetCamera();
+      Module.engineInstance.getWindow().render();
+    };
+
+    // setup file open event
+    const progressEl = document.querySelector("#progressEl");
+    const fileSelector = document.querySelector("#file-selector");
+    fileSelector.addEventListener("change", (evt) => {
+      for (const file of evt.target.files) {
+        const reader = new FileReader();
+        reader.addEventListener("loadend", (e) => {
+          progressEl.textContent = "100%";
+          progressEl.value = 100;
+          Module.FS.writeFile(file.name, new Uint8Array(reader.result));
+          openFile(file.name);
+          progressEl.parentElement.style.display = "none";
+        });
+        reader.addEventListener("progress", (evt) => {
+          console.log(evt);
+          const progress = Math.floor((100 * evt.loaded) / evt.total);
+          progressEl.parentElement.style.display = "block";
+          progressEl.value = progress;
+          progressEl.textContent = `${progress}%`;
+        });
+        reader.readAsArrayBuffer(file);
+      }
+    });
+
+    Module.setupOptions(Module.engineInstance.getOptions());
+
+    // Storing dom el ids to f3d option mappings since also useful for url-param parsing
```

Comment:
dom el ?

Reply:
DOM element. I've updated the comment

Reply:
Keep in mind this file is a copy/paste of the content that was in the old app.html

Reply:
which I probably not reviewed thoroughly ^^